### PR TITLE
Jsonize test data for vehicle drag and effective dps tests

### PR DESCRIPTION
**Pull request by mqrause**
#### Summary
Infrastructure "Jsonize test data for vehicle drag and effective dps tests"

#### Purpose of change

These tests regularly cause confusion in json contributors. Jsonizing the data means they don't have to touch c++ files anymore.

#### Describe the solution

Add a class for test data with static members to be accessed in the tests.
Add members for vehicle drag and effective dps and also move the known bad density data there.
Move the data from the tests to json.

Adding data for other tests is easy.

#### Testing

Explicitly ran the affected tests and all loaded their data and passed.

### CODE REVIEW

File: src/itype.h [hunk 1]
```diff
@@ -1411,12 +1411,5 @@ struct itype {
 
 void load_charge_removal_blacklist( const JsonObject &jo, const std::string &src );
 void load_charge_migration_blacklist( const JsonObject &jo, const std::string &src );
-// can be removed once all known bad items got fixed
-class known_bad_density
```
Comment (Leland) [inline src/itype.h:1415]: I'll note that we are losing some context from comments in a few places.

Comment (mqrause) [reply src/itype.h:1415]: Yeah, I judged the comment isn't really relevant enough when it's not it's own class anymore. I can readd it if people think it should still be there.

**Comment by mqrause** [comment]
Oh yeah, forgot about that. I was also considering doing a full object instead of the array, but I'm definitely too lazy make the json for that.

**Comment by ZhilkinSerg** [comment]
> Note on the vehicle test data json on what the numbers mean would be nice.

Test data is generated automatically, so it should not matter much.

File: src/itype.h (hunk 1 above)
Comment (mqrause) [reply src/itype.h:1415]: Added a comment.

**Pull request merged**